core/sql/optimizer/OptPhysRelExpr.cpp

```diff
@@ -3910,6 +3910,20 @@ InputPhysicalProperty* NestedJoin::generateIpp(
   InputPhysicalProperty* ipp = NULL;
 
   // ----------------------------------------------------------------
+  // If there is a cardinality constraint on the outer child, we'll
+  // pass that information along to the inner child.
+  // ----------------------------------------------------------------
+  Cardinality minRows;
+  Cardinality maxRows;
+  CardConstraint * outerCardinalityConstraint = NULL;
+
+  if (getGroupAttr()->hasCardConstraint(minRows, maxRows))
```

Comment:
Yes, the intent of the change is to send any cardinality constraint on the result of the left child down to the right child scan as a constraint on the number of probes. Before this change, no such constraint was passed down; we relied on estimates only. The difference is a matter of certainty. MDAM was considered dangerous to do on the inner child of a nested join due to unreliability of estimates in some plans.